### PR TITLE
Remove unused `--minimum-buffer-size` option

### PR DESCRIPTION
**Pull request by ZimbiX**
It became unused in https://github.com/que-rb/que/commit/1736d3a30612f90a6839c5f98f622c9585e51784, so since v1.0.0.beta4.

It's just confusing things now, e.g. in:

- https://github.com/que-rb/que/issues/345
- https://github.com/que-rb/que/issues/331

### CODE REVIEW

File: app.rb
Comment (oeoeaio) [inline app.rb:15]: Don’t think we need the app. 😆

Comment (ZimbiX) [reply app.rb:15]: Hah! Whoops

**Review by oeoeaio — APPROVED** [review]
Looks good but probably want to merge this into a v2 branch or after we merge and release the ruby3 stuff because it’s a breaking change.

Edit: or will the CLI just ignore the old minimum buffer size option?

**Comment by ZimbiX** [comment]
It rejects the option now:

```
➜ bundle exec que --minimum-buffer-size 0 ./app.rb                                                                                                                                                                                                                          bundler: failed to load command: que (/home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/bin/que)
Traceback (most recent call last):
	18: from /home/brendan/.rbenv/versions/2.7.3/bin/bundle:25:in `<main>'
	17: from /home/brendan/.rbenv/versions/2.7.3/bin/bundle:25:in `load'
	16: from /home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/exe/bundle:36:in `<top (required)>'
	15: from /home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
	14: from /home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/exe/bundle:48:in `block in <top (required)>'
	13: from /home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/cli.rb:25:in `start'
	12: from /home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	11: from /home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/cli.rb:31:in `dispatch'
	10: from /home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	9: from /home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	8: from /home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	7: from /home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/cli.rb:484:in `exec'
	6: from /home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/cli/exec.rb:23:in `run'
	5: from /home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/cli/exec.rb:58:in `kernel_load'
	4: from /home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.3.7/lib/bundler/cli/exec.rb:58:in `load'
	3: from /home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/bin/que:25:in `<top (required)>'
	2: from /home/brendan/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/bin/que:25:in `load'
	1: from /home/brendan/Projects/que/bin/que:9:in `<top (required)>'
/home/brendan/Projects/que/bin/command_line_interface.rb:148:in `parse': invalid option: --minimum-buffer-size (OptionParser::InvalidOption)
Did you mean?  maximum-buffer-size
```

I considered ignoring it, but thought it'd just be fine to remove with a minor version bump?

**Comment by ZimbiX** [comment]
Hmm, sounds like that's [not good](https://semver.org/):

> MINOR version when you add functionality in a backwards compatible manner

**Comment by ZimbiX** [comment]
Ok, I've made it just be ignored with a deprecation warning. We can remove it in v2.0.

**Pull request merged**
